### PR TITLE
 Bug Report: User unable to join Organizations error solved

### LINKED ISSUE #1849: Bug Report: User unable to join Organizations
**Describe the bug**
User unable to join the organization.

**Expected behavior**
user should be able to send the request.

**Screenshots**
current situation:
image

**Desired Result:**
In the User portal the user will want to `join` the organization
image

**Past similar issue:** #1511 

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by gautam-divyanshu**
**What kind of change does this PR introduce?**
User will able to join the organization.

**Issue Number:**

Fixes #1849 

**Did you add tests for your changes?**

**Snapshots/Videos:**
Screenshot from 2024-04-04 20-14-24

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

### CODE REVIEW

File: src/screens/UserPortal/Organizations/Organizations.tsx [hunk 1]
```diff
@@ -101,6 +107,7 @@ export default function organizations(): JSX.Element {
       filter: value,
     });
   };
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (pranshugupta54) [inline src/screens/UserPortal/Organizations/Organizations.tsx:110]: don't disable eslint and fix them

Comment (gautam-divyanshu) [reply src/screens/UserPortal/Organizations/Organizations.tsx:110]: Ok

File: src/screens/UserPortal/Organizations/Organizations.tsx [hunk 2]
```diff
@@ -101,6 +107,7 @@ export default function organizations(): JSX.Element {
...
   const handleSearchByEnter = (e: any): void => {
```
Comment (pranshugupta54) [inline src/screens/UserPortal/Organizations/Organizations.tsx:111]: Try using 
```ts
const handleSearchByEnter = (e: React.KeyboardEvent<HTMLInputElement>): void => {
  if (e.key === 'Enter') {
    const { value } = e.currentTarget;
    handleSearch(value);
  }
};
```

**Review by palisadoes — CHANGES_REQUESTED** [review]
Add a test for the bug. We don't want this to happen again.

**Comment by gautam-divyanshu** [comment]
@palisadoes currently there is no page for request handling in Admin portal 

[Screencast from 2024-04-04 23-47-03.webm](https://github.com/PalisadoesFoundation/talawa-admin/assets/137816099/780d5e73-ddd7-42c3-9b2a-557956096e4e)

**Comment by gautam-divyanshu** [comment]
@palisadoes Should I cover that issue in my issue?

**Comment by pranshugupta54** [comment]
@gautam-divyanshu, it's there before selecting an Org. (If there should be an improvement in it, then please take it separately to avoid a very long PR of 2 different issue types)
image

**Comment by gautam-divyanshu** [comment]
Oh, the problem is that I was logged in as a superadmin, and when I try to access /requests, the Request page does not load.

**Comment by palisadoes** [comment]
1. Requests should be handled at the Admin level.
2. The tab needs to be moved from SuperAdmin.
3. Only Organization specific requests should be visible in the Admin panel.
4. This should be covered in the official Figma design

**Comment by gautam-divyanshu** [comment]
@palisadoes  ok, I will remove the request button from superAdminScreen

**Comment by pranshugupta54** [comment]
@gautam-divyanshu, I've made a separate issue for it. We should not mix independent issues of user portal and admin portal in one PR

**Comment by gautam-divyanshu** [comment]
> @gautam-divyanshu, I've made a separate issue for it. We should not mix independent issues of user portal and admin portal in one PR

I have updated the code, it requires just minor issues. 
@palisadoes you should suggest what should be done

**Comment by palisadoes** [comment]
Don't change the original scope of this PR.

The other issue can be addressed later.

**Comment by gautam-divyanshu** [comment]
[Screencast from 2024-04-05 21-42-23.webm](https://github.com/PalisadoesFoundation/talawa-admin/assets/137816099/fade8b4a-0724-4433-b6b2-afaa7e7c2767)

@palisadoes We can close this issue , #1874
I made changes for that in this pr.

**Comment by palisadoes** [comment]
Please fix the linting & tests

**Comment by gautam-divyanshu** [comment]
@palisadoes Please review

**Comment by gautam-divyanshu** [comment]
@palisadoes  And can we disable this workflow approval process, just to get things fast

**Comment by palisadoes** [comment]
1. The approvals are only required for your very first PR in a repo. The easiest way is to create another PR before this one so that this one won't require any more approvals.
2. Do a simple PR against the same issue. A minor change in INSTALLATION.md, like add a space. When that is merged, the problem with this PR will go away

**Comment by gautam-divyanshu** [comment]
@palisadoes  please approve

**Comment by palisadoes** [comment]
Subsequent commits should automatically trigger the GitHub action's

**Comment by gautam-divyanshu** [comment]
Then what might be use?

**Comment by palisadoes** [comment]
Please fix the failing tests

**Comment by gautam-divyanshu** [comment]
@palisadoes these test cases were failing even before my pr

**Comment by palisadoes** [comment]
This is the most recently merged PR. All its tests passed. Please merge with the latest upstream then verify

- https://github.com/PalisadoesFoundation/talawa-admin/pull/1878

**Review by palisadoes — CHANGES_REQUESTED** [review]
Please get the test code coverage of the patch higher. It's only 11%

**Comment by gautam-divyanshu** [comment]
@palisadoes  I will, but currently I will not be able to because of health issues.

**Comment by pranshugupta54** [comment]
Test is failing in un-changed files bcuz you changed Query which is indirectly affecting other files

**Review by palisadoes — APPROVED** [review]

**Comment by palisadoes** [comment]
@gautam-divyanshu 

My 11% code coverage request was for another PR. I used the mobile app and I've noticed that it has a bug where it sometimes posts comments in the previously viewed PR especially after reviewing a lot in sequence.

Now I can't dismiss it.

**Comment by gautam-divyanshu** [comment]
@palisadoes what should I do?

**Comment by palisadoes** [comment]
See comment 

- https://github.com/PalisadoesFoundation/talawa-admin/pull/1875#issuecomment-2041101097

**Comment by pranshugupta54** [comment]
@gautam-divyanshu, any updates? Are you facing some issue? This is a major issue which stops User Portal from main functionality.

**Comment by gautam-divyanshu** [comment]
> @gautam-divyanshu, any updates? Are you facing some issue? This is a major issue which stops User Portal from main functionality.

@pranshugupta54 I will make the updates soon, I was inactive because of health issues

**Comment by pranshugupta54** [comment]
Ohk no problem, take care! If you need help, feel free to connect on Slack.

**Comment by gautam-divyanshu** [comment]
@palisadoes  Please review

**Comment by pranshugupta54** [comment]
Why didn't we get an error on `src/components/UserPortal/OrganizationCard/OrganizationCard.test.tsx` which checks "Join Now" button 🤔

**Comment by gautam-divyanshu** [comment]
> mponents/UserPortal/OrganizationCard/OrganizationCa

This check is performed using the `expect(screen.getByTestId('joinBtn')).toBeInTheDocument()`

**Comment by pranshugupta54** [comment]
Yeah but how did it pass when there was no button on screen??

**Comment by gautam-divyanshu** [comment]
> Yeah but how did it pass when there was no button on screen??

The tests passed because we focused on the underlying functionality triggered by a mocked element (button with data-testid) rather than its physical presence on the screen.

**Comment by pranshugupta54** [comment]
data-testid won't be there if button doesn't render

**Comment by gautam-divyanshu** [comment]
> data-testid won't be there if button doesn't render

If the "Join Now" button is not rendered due to incorrect conditions or missing data, the screen.getByTestId('joinBtn') call will fail to find the button, and the test will also fail.

**Comment by pranshugupta54** [comment]
"Join Now" didn't render thats why we made an issue and this is the PR but the test was passing earlier too, means there is some issue with test so you should fix it in this PR itself to completely solve this and not get this issue in future again.

**Comment by gautam-divyanshu** [comment]
> "Join Now" didn't render thats why we made an issue and this is the PR but the test was passing earlier too, means there is some issue with test so you should fix it in this PR itself to completely solve this and not get this issue in future again.

I have added a test

**Comment by gautam-divyanshu** [comment]
@palisadoes  Please review

**Review by palisadoes — APPROVED** [review]

**Pull request merged**